cli/provider/cmd.go

```diff
@@ -523,3 +518,88 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	}
 	return nil
 }
+
+func PreProcessCoverage(logger *zap.Logger, conf *config.Config) {
```

Comment:
maybe good to move a good chunk of this logic to the coverage package

Reply:
done!